cli/cmd/proxy/read/command.go

```diff
@@ -258,19 +276,19 @@ func (c *ReadCommand) outputConfig(config *EnvoyConfig) error {
 
 func (c *ReadCommand) outputAsJSON(config *EnvoyConfig) error {
 	cfg := make(map[string]interface{})
-	if !c.filtersPassed() || c.flagClusters {
-		cfg["clusters"] = config.Clusters
+	if !c.areTablesFiltered() || c.flagClusters {
```

Comment:
this logic seems pretty complicated when you look at at the `areTablesFiltered` implementation. It might worth adding a comment here explaining why we're doing it.

Reply:
I agree. I finally found a solution I like which is to have an `shouldPrintTable` method that makes the whole thing clearer.